pkg/driver/driver_openstack.go

```diff
@@ -160,6 +161,29 @@ func (d *OpenStackDriver) Create() (string, string, error) {
 
 	metrics.APIRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "nova"}).Inc()
 
+	var useConfigDrive *bool
```

Comment:
I'd move the logic below into a separate func. However the whole `Create` func already consists of `if then else`. Apart from the above, it's better to define `var useConfigDrive bool` here and use `ConfigDrive:      &useConfigDrive,` in `createOpts definition.